src/ex4/server/services/item_manager.js

```diff
@@ -1 +1,89 @@
-// The ItemManager should go here. Remember that you have to export it.
\ No newline at end of file
+const fs = require('fs').promises;
+const tasksFile = 'tasks_file.json';
+const pokemonApi = require('../clients/pokemon_client');
+const pokemonClient = new pokemonApi();
+const cache = new Map();
+
+async function addTasksOrPokemons(tasks) {
+  let data = await readTasksFile();
+  if (!data) {
+    data = [];
+  }
+
+  const cachedTasks = tasks.filter((task) => !isNaN(task) && cache.has(task)); // find cached tasks (only numbers)
+
+  const result = await pokemonClient.fetchPokemons(
+    tasks.filter((item) => !cache.has(item))
+  ); // fetch only new requests
```

Comment:
try to think how to remove those comments and make the code more readable. in this case, you could of create a `uncachedTasks` list, and use it here instead.
```suggestion
  const uncacheTasks = tasks.filter((item) => !cache.has(item))
  const result = await pokemonClient.fetchPokemons(
    uncachedTasks
  );
```